src/components/Gradebook/index.jsx

```diff
@@ -515,6 +517,32 @@ export default class Gradebook extends React.Component {
     return valueAsInt >= 0 && valueAsInt <= 100;
   };
 
+  handleFilterBadgeClose = filterNames => () => {
+    this.props.resetFilters(filterNames);
+    const queryParams = {};
+    filterNames.forEach((filterName) => {
+      queryParams[filterName] = false;
+    });
+    this.updateQueryParams(queryParams);
+    const stateUpdate = {};
+    const rangeStateFilters = ['assignmentGradeMin', 'assignmentGradeMax', 'courseGradeMin', 'courseGradeMax'];
+    rangeStateFilters.forEach((filterName) => {
+      if (filterNames.includes(filterName)) {
+        stateUpdate[filterName] = initialFilters[filterName];
+      }
+    });
+    this.setState(
+      stateUpdate,
+      () => {
+        this.props.getUserGrades(
```

Comment:
I don't think there's actually a compelling reason these can't happen in parallel, which should help free you up to pull more of this `handleFilterBadgeClose`'s logic into the subcomponent.

Reply:
Yeah, you're right. the setState is just for updating the displayed values on the page, not the actual filter values, so I should be good... the filter values in the props should have been updated by resetFilters already, right?

Reply:
Yeah from what I can see as long as we dispatch what's behind `getUserGrades` strictly after we dispatch the action that clears the filters, we should be good.